### DISCUSSION TITLE
How to add attachment to unhandled exception in MAUI?

### DISCUSSION THREAD

**Discussion by lubiepomaranczki** [getsentry/sentry-dotnet#3346] [Q&A] [answered]
Hello there,
I am playing a bit with Sentry to check if we can replace AppCenter with Sentry.

I am wondering what is the best approach to add Attachment to unhandled exception?

Let's say that once unhandled exception happens, I would like to add additional logs as attachment. 

I found out, that you could configure scope globally during setup
```
.UseSentry(options =>
{
    options.ConfigureScope(scope =>
    {
        var bytes = GetMyLogs();
        if (bytes != null)
        {
            var attachment = new SentryAttachment(
                AttachmentType.Default, 
                new ByteAttachmentContent(bytes),
                $"mobile_log_{Guid.NewGuid()}.zip", 
                "application/zip");

            scope.AddAttachment(attachment);
        }
    });
```
but this is not ideal, as those are added every time sentry is being used. What's more, the logs from the current session wan I also didn't find way to filter this only for exceptions.

EDIT: It happened that this is not the way as this is only configured on startup. Hence, this will get latest logs and add those as attachment. When the error happens, this **won't get latest logs** but it will use those from startup.

How do you do this?

**Comment by bitsandfoxes** [comment, marked as answer]
Hey there! I think you're looking for the [`BeforeSend`](https://docs.sentry.io/platforms/dotnet/configuration/filtering/#using-) callback. The callback provides a `Hint` object that takes paths as attachments that then get added to the event during later stages of eventprocessing.
It'd look a bit like this
```
.UseSentry(options =>
{
    options.SetBeforeSend((sentryEvent, hint) =>
    {
        hint.AddAttachment("path/to/my/logs.log");
        return sentryEvent;
    });
});
```

**Reply by lubiepomaranczki** [reply]
That's a perfect tip! Thanks a lot! 🎉
